payments-ui-core/src/main/java/com/stripe/android/ui/core/forms/resources/LpmRepository.kt

```diff
@@ -393,17 +393,24 @@ class LpmRepository constructor(
             requirement = AuBecsDebitRequirement,
             formSpec = LayoutSpec(sharedDataSpec.fields)
         )
-        PaymentMethod.Type.USBankAccount.code -> SupportedPaymentMethod(
-            code = "us_bank_account",
-            requiresMandate = true,
-            displayNameResource = R.string.stripe_paymentsheet_payment_method_us_bank_account,
-            iconResource = R.drawable.stripe_ic_paymentsheet_pm_bank,
-            lightThemeIconUrl = sharedDataSpec.selectorIcon?.lightThemePng,
-            darkThemeIconUrl = sharedDataSpec.selectorIcon?.darkThemePng,
-            tintIconOnSelection = true,
-            requirement = USBankAccountRequirement,
-            formSpec = LayoutSpec(sharedDataSpec.fields)
-        )
+        PaymentMethod.Type.USBankAccount.code -> {
```

Comment:
(not as part of this PR), should we make this into a factory pattern? We create an interface, pass in the code (and whatever other information we need) to a function, we return a nullable `SupportedPaymentMethod?`, then we can add these factories to a list, and loop through them. Or we could add them to a map with the code being the key.

There's getting to be more and more logic in each of these, and it would be nice to test them in isolation, and maybe generate some helpers to make the creation of them more consistent.

Reply:
Yep, I broadly agree with these ideas.